account_payment_term_extension/models/res_company.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2024 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    payment_terms_delay_type = fields.Selection(
+        string="Payment terms Delay types",
+        required=True,
+        default="days",
+        selection=[
+            ("days", "Days"),
+            ("weeks", "Days and Weeks"),
+            ("months", "Days and Months"),
+            ("weeks_and_months", "Days, Weeks and Months"),
```

Comment:
Why weeks_and_months is not the default? In v16 you dont have to edit the settings to use this feature

Reply:
In previous versions like v16, Odoo included months and weeks in addition to days by default. However, in the v17, the default has been simplified to only include days. Given that the module extends the base functionality, the days are chosen as the default setting to keep consistency with the core functionality.